src/libs/Violations/ViolationsUtils.ts

```diff
@@ -0,0 +1,69 @@
+import reject from 'lodash/reject';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import {PolicyCategories, PolicyTags, Transaction, TransactionViolation} from '@src/types/onyx';
+import possibleViolationsByField, {ViolationField} from './possibleViolationsByField';
+
+const ViolationsUtils = {
+    getViolationForField(transactionViolations: TransactionViolation[], field: ViolationField, translate: (key: string) => string): string[] {
+        return transactionViolations.filter((violation) => possibleViolationsByField[field]?.includes(violation.name)).map((violation) => translate(violation.name));
```

Comment:
I'm actually not a huge fan of the extra complexity added by `possibleViolationsByField`, and I suggest we get rid of it.

In practice, it'll turn this line into something like this, which is just as simple
```js
return transactionViolations.filter((violation) => violationFields[violation.name] === field).map((violation) => translate(violation.name));
```

Let's also get rid of `src/libs/Violations/possibleViolationsByField.ts` and move that logic into this file

Reply:
Yeah this was all a bit clunky, but as I used it in components I found a better solution. I was going to include it as part of my  PR on the MoneyRequestView issue, but I will bring the changes into this branch so we can look at it here.